Sources/Tools/SoundDecompress/plOldAudioFileReader.cpp

```diff
@@ -0,0 +1,177 @@
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+//////////////////////////////////////////////////////////////////////////////
+//                                                                          //
+//  plOldAudioFileReader                                                    //
+//                                                                          //
+//// Notes ///////////////////////////////////////////////////////////////////
+//                                                                          //
+//  3.5.2001 - Created by mcn.                                              //
+//                                                                          //
+//////////////////////////////////////////////////////////////////////////////
+
+#include "HeadSpin.h"
+#include "plOldAudioFileReader.h"
+//#include "hsTimer.h"
+
+#include "plUnifiedTime/plUnifiedTime.h"
+#include "plAudioCore/plBufferedFileReader.h"
+#include "plAudioCore/plCachedFileReader.h"
+#include "plAudioCore/plFastWavReader.h"
+#include "plAudioCore/plOGGCodec.h"
+#include "plAudioCore/plWavFile.h"
+
+#define kCacheDirName   "streamingCache"
+
+plAudioFileReader* plOldAudioFileReader::CreateReader(const plFileName& path, plAudioCore::ChannelSelect whichChan, StreamType type)
+{
+    ST::string ext = path.GetFileExt();
+
+    if (type == kStreamWAV)
+    {
+        bool isWav = (ext.compare_i("wav") == 0);
+        // We want to stream a wav off disk, but this is a compressed file.
+        // Get the uncompressed path. Ignore the requested channel, since it 
+        // will have already been split into two files if that is necessary.
+        if (!isWav)
+        {
+            plFileName cachedPath = IGetCachedPath(path, whichChan);
+            plAudioFileReader *r =  new plFastWAV(cachedPath, plAudioCore::kAll);
+            if (!r->IsValid()) {
+                // So we tried to play a cached file and it didn't exist
+                // Oops... we should cache it now
+                delete r;
+                ICacheFile(path, true, whichChan);
+                r = new plFastWAV(cachedPath, plAudioCore::kAll);
+            }
+            return r;
+        }
+
+        plAudioFileReader *r =  new plFastWAV(path, whichChan);
+        return r;
+    }
+    else if (type == kStreamRAM)
+        return new plBufferedFileReader(path, whichChan);
+    else if (type == kStreamNative)
+        return new plOGGCodec(path, whichChan);
+
+    return nil;
+}
+
+plAudioFileReader* plOldAudioFileReader::CreateWriter(const plFileName& path, plWAVHeader& header)
+{
+#ifdef HS_BUILD_FOR_WIN32
+    plAudioFileReader* writer = new CWaveFile(path.AsString().c_str(), plAudioCore::kAll);
+    writer->OpenForWriting(path.AsString().c_str(), header);
+    return writer;
+#else
+    hsAssert(false, "WAV files can only be generated on Windows");
```

Comment:
Given that wav file production is a compile time option, this should be a compile time error with `static_assert` instead of a runtime error.